Upgrade test: test peering upgrade from an old version of consul

huikang opened the pull request:
### Description
- Upgrade test case for peering: verify peering state can be recovered to `ACTIVE` after consul agent is upgraded to the new version (`TestPeering_UpgradeToTarget_fromLatest`)
- container logs are disabled because in my local test, I observed some flakiness in follow logs in testcontainers. A new flag `follow-log` is added to enable following the log in stdout

### Testing & Reproduction steps

### Links

### PR Checklist

* [x] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

DanStough commented on test/integration/consul-container/libs/agent/container.go:
```diff
@@ -199,20 +204,24 @@ func (c *consulContainerNode) Exec(ctx context.Context, cmd []string) (int, erro
 		return err
 	}
 
-	consulType := "client"
-	if pc.Server {
-		consulType = "server"
-	}
-	name := utils.RandName(fmt.Sprintf("%s-consul-%s-%d", pc.Datacenter, consulType, index))
+	// NOTE: keep this as we may want to add an option to change the node name in certain scenario
+	// consulType := "client"
+	// if pc.Server {
+	// 	consulType = "server"
+	// }
+	// name := utils.RandName(fmt.Sprintf("%s-consul-%s-%d", pc.Datacenter, consulType, c.id))
```
IMHO we just delete all this, including readSomeConfigFileFields, to keep it cleaner.

DanStough commented on test/integration/consul-container/test/upgrade/peers_http_test.go:
```diff
@@ -0,0 +1,91 @@
+	dialingPeerName   = "dialing-to-acceptor"
+)
+
+func TestPeering_UpgradeToTarget_fromLatest(t *testing.T) {
```
Not a dealbreaker, but I think it's helpful to add some context about why this test case exists or what it is testing in the comments.

DanStough reviewed (commented):
Comments more than suggestions. Feel free to push back on any of them.

With all the refactoring, it's awesome how concise the new test case is 🥳 !

huikang commented on test/integration/consul-container/test/basic/connect_service_test.go:
```diff
@@ -25,7 +25,7 @@ func TestBasicConnectService(t *testing.T) {
 	defer terminate(t, cluster)
 
-	clientService := createServices(t, cluster)
+	clientService := createServices(t, cluster, *utils.FollowLog)
```
Agree. Will update. You feedback is very helpful. Thanks.

huikang commented on test/integration/consul-container/libs/utils/version_oss.go:
```diff
@@ -12,3 +12,4 @@ var TargetImage = flag.String("target-image", "consul", "docker image name to be
 var TargetVersion = flag.String("target-version", "local", "docker image version to be used as UUT (unit under test)")
 var LatestImage = flag.String("latest-image", "consul", "docker image name to be used under test (Default: consul)")
 var LatestVersion = flag.String("latest-version", "1.11", "docker image to be used as latest")
+var FollowLog = flag.Bool("follow-log", false, "follow container log in output")
```
Thanks. I will create a follow-up task to import the upgrade test to ENT

huikang replied:
@DanStough , just want to make sure I understand you correct. Are you saying the ` compatibility-integration-test:` in ENT still uses the OSS image (the link is omitted for privacy)? Please correct me if I am wrong. Thanks.

DanStough replied:
I just wanted to make sure this line gets added to `test/integration/consul-container/libs/utils/version_ent.go` in case we want to run enterprise-specific tests. It just so happens it isn't different from this file and I don't think we use it. TBH we could just get rid of the build tags and rename this file `version.go`.

huikang replied:
Got it, @DanStough . 

@eculver  is working on enabling the upgrade test in ENT and we will pay attention to this. Thanks.

huikang commented:
Hi, @DanStough , I have updated the PR addressing most of your review comments. When you have a chance, please take another look and let me know what I might be missing. Thanks.
